### PR TITLE
Remove "Draw box to filter" from grid maps for native queries

### PR DESCRIPTION
**Pull request by ranquild**
How to verify:
- New -> SQL query -> `SELECT * FROM PEOPLE LIMIT 10` -> Save
- Change the viz type to Map
- Viz settings -> Map type -> Grid map. Select columns (latitude, longitude) -> **Save**
- Make sure that `Draw box to filter` isn't available
- Viz settings -> Map type -> Pin map -> **Save**
- Make sure that `Draw box to filter` is available

Grid map:
Screenshot 2024-10-08 at 12 39 35

Pin map:
Screenshot 2024-10-08 at 12 41 10

### CODE REVIEW

File: frontend/src/metabase/visualizations/visualizations/CartesianChart/events.ts [hunk 1]
```diff
@@ -273,6 +275,7 @@ export const canBrush = (
     !!onChangeCardAndRun &&
     hasBrushableDimension &&
     !hasCombinedCards &&
+    (!isNative(series[0].card) || isSavedCard(series[0].card)) &&
```
Comment (ranquild) [inline frontend/src/metabase/visualizations/visualizations/CartesianChart/events.ts:278]: Leftover from https://github.com/metabase/metabase/pull/48232/files#diff-5c077bf5973eb7d1479a32f2c0d0ab1ba5c9da135a18f4a9832b9b82a4237753L277

File: frontend/src/metabase/visualizations/click-actions/Mode/Mode.ts [hunk 2]
```diff
@@ -37,37 +37,31 @@ export class Mode {
     settings: Record<string, any>,
     extraData?: Record<string, any>,
   ): ClickAction[] {
-    // keeping try/catch here as MBQL was failing silently
```
Comment (ranquild) [inline frontend/src/metabase/visualizations/click-actions/Mode/Mode.ts:40]: Reverts changes from the original PR

Comment (uladzimirdev) [reply frontend/src/metabase/visualizations/click-actions/Mode/Mode.ts:40]: well, maybe it's not needed, but mbql was failing silently without it

File: frontend/src/metabase/visualizations/components/PinMap.jsx [hunk 3]
```diff
@@ -255,33 +255,38 @@ export default class PinMap extends Component {
...
-                ) {
-                  this._map.stopFilter();
-                }
-              }}
-            >
-              {!this.state.filtering ? t`Draw box to filter` : t`Cancel filter`}
-            </div>
-          )}
+          {!isDashboard &&
+            this._map &&
```
Comment (uladzimirdev) [inline frontend/src/metabase/visualizations/components/PinMap.jsx:259]: `this._map?.supportsFilter?.()`?

**Review by uladzimirdev — APPROVED** [review]

**Pull request merged**